rkyv_wrappers/src/custom_phantom.rs

```diff
@@ -1,56 +1,71 @@
 //! A wrapper that converts a `Vec` to an `ArchivedHashMap` at serialization time.
 
 use rkyv::{
+    rancor::Fallible,
     with::{ArchiveWith, DeserializeWith, SerializeWith},
-    Fallible
+    Place,
 };
 use std::marker::PhantomData;
 
 /// A wrapper that allows for changing the generic type of a PhantomData<T>.
-/// 
+///
 /// Example:
-/// 
+///
 /// ```rust
 /// use std::marker::PhantomData;
 /// use rkyv::{
-///     Archive, Serialize, Deserialize, Infallible, vec::ArchivedVec, Archived, with::With,
+///     deserialize, bytecheck, Portable, Archive, Deserialize, Serialize, rancor::Infallible, vec::ArchivedVec, Archived, with::With,
+///     rancor::Error,
 /// };
 /// use rkyv_wrappers::custom_phantom::CustomPhantom;
-/// #[derive(Archive, Serialize, Deserialize, Debug, PartialEq, Eq, Default)]
-/// #[archive(as = "StructWithPhantom<T::Archived>", bound(archive = "
-/// 	T: Archive,
```

Comment:
I figured T doesn't really need to be `Archive`, as `CustomPhantom` is `Archive` ; it was there only for `as = "StructWithPhantom<T::Archived>"`, which is impossible now that i32 is not `Portable`.

We could make this example using `as` if i32 were generic over `aDifferentT: Archive`, but that's controversial and I encountered errors. I'd still like to see such example (especially helpful in context of my nalgebra migration).